frontend/src/metabase/querying/notebook/components/ClauseStep/ClauseStep.module.css

```
@@ -0,0 +1,4 @@
+.closeIcon {
+  margin-left: 0.5rem;
+  flex: 0 0 auto;
```

nit, should be enough

```suggestion
  flex-shrink: 0;
```